library/src/main/java/com/datatorrent/lib/appdata/gpo/GPOUtils.java

```diff
@@ -123,6 +123,54 @@ public static GPOMutable deserialize(FieldsDescriptor fieldsDescriptor,
   }
 
   /**
+   * This method deserializes the fields in the given {@link FieldsDescriptor} into a map.
+   * @param fieldsDescriptor The {@link FieldsDescriptor} to fetch fields from.
+   * @param dpou The {@link JSONObject} which contains the fields whose values need to be fetched.
+   * @return A {@link Map} whose keys are field names, and whose values are possible values for those fields.
+   */
+  public static Map<String, Set<Object>> deserializeToMap(FieldsDescriptor fieldsDescriptor,
+                                                          JSONObject dpou)
+  {
+    Map<String, Set<Object>> keyToValues = Maps.newHashMap();
+
+    for (String key : fieldsDescriptor.getFields().getFields()) {
+      if (!dpou.has(key)) {
+        throw new IllegalArgumentException("The given key " + key + " is not contained in the given JSON");
+      }
+
+      Type type = fieldsDescriptor.getType(key);
+      Set<Object> keyValues = Sets.newHashSet();
+      keyToValues.put(key, keyValues);
+
+      Object keyValue;
+
+      try {
+        keyValue = dpou.get(key);
+      } catch (JSONException ex) {
```

Comment:
This is util class and it should not catch exceptions. It should be left to user to swallow it or rethrow it. Now with your changes user doesn't have that control.